lib/rspec/mocks/order_group.rb

```diff
@@ -66,15 +71,27 @@ def expectations_invoked_in_order?
       end
 
       def invoked_expectations
-        @expectations.select { |e| e.ordered? && @invocation_order.include?(e) }
+        @expectations[order_range].select { |e| e.ordered? && @invocation_order.include?(e) }
       end
 
       def expected_invocations
-        @invocation_order.map { |invocation| expectation_for(invocation) }.compact
+        @invocation_order[order_range].map { |invocation| expectation_for(invocation) }.compact
       end
 
       def expectation_for(message)
-        @expectations.find { |e| message == e }
+        @expectations[order_range].find { |e| message == e }
+      end
+
+      def order_range
+        (start_range..end_range)
+      end
+
+      def start_range
+        @index - (@index.zero? ? 0 : 1)
+      end
+
+      def end_range
```

Comment:
I'd normally expect any method that ends in `_range` to return a range.  Could this be `range_start` and `range_end` as those names make it more clear that these methods don't return ranges but rather the start and end of one?